packages/core/core/src/public/Bundle.js

```diff
@@ -282,7 +283,26 @@ export class PackagedBundle extends NamedBundle implements IPackagedBundle {
     return packagedBundle;
   }
 
+  static getWithInfo(
+    internalBundle: InternalBundle,
+    bundleGraph: BundleGraph,
+    options: ParcelOptions,
+    bundleInfo: ?PackagedBundleInfo,
+  ): PackagedBundle {
+    let packagedBundle = PackagedBundle.get(
+      internalBundle,
+      bundleGraph,
+      options,
+    );
+    packagedBundle.#bundleInfo = bundleInfo;
+    return packagedBundle;
+  }
+
   get filePath(): string {
-    return nullthrows(this.#bundle.filePath);
+    return nullthrows(this.#bundleInfo).filePath;
```

Comment:
Previously, we mutated the bundle after packaging to set the file path we actually wrote to. This has a number of problems with caching, so I moved this to a separate bundle info object. It's transparent to the end user, but internally, this is not stored on the actual bundle node.